torch/package/package_importer.py

```diff
@@ -27,8 +23,29 @@
 from .file_structure_representation import Directory, _create_directory_from_file_list
 from .glob_group import GlobPattern
 from .importer import Importer
-from ._zip_file import PackageZipFileReader
-from ._zip_file_torchscript import TorchScriptPackageZipFileReader
+from ._zip_file import PackageZipFileReader, DefaultPackageZipFileReader
+
+
+def import_torch():
+    global torch
+    global _get_restore_location
+    global TorchScriptDirectoryReader
+    global TorchScriptPackageZipFileReader
+    import torch
+    from ._directory_reader_torchscript import TorchScriptDirectoryReader
+    from ._zip_file_torchscript import TorchScriptPackageZipFileReader
+    from torch.serialization import _get_restore_location
```

Comment:
these conditional global imports get quite fragile, if we can wrap them with fallbacks at import time that seems much better